MoltenVK/MoltenVK/Commands/MVKCommandEncoderState.mm

```diff
@@ -402,7 +402,8 @@
 #pragma mark MVKResourcesCommandEncoderState
 
 // Updates the swizzle for an image in the given vector.
-static void updateSwizzle(MVKVector<uint32_t> &constants, uint32_t index, uint32_t swizzle) {
+template<typename T>
+static void updateSwizzle(T &constants, uint32_t index, uint32_t swizzle) {
```

Comment:
This should no longer need to be a template. In fact, you shouldn't need to change this at all anymore.

Reply:
@cdavis5e What do you mean by "shouldn't need to change this at all"?

Reply:
Hmmm...attempting to change this back from a template causes compile errors. Wasn't obvious to me why...so I'm leaving it for now.

Reply:
That is, it should've been possible to just use `MVKVector<uint32_t>&` as the type of `constants`.

Reply:
Sorry for not commenting on my change here. If you use MVKVector as the function parameter, this means you are accessing the vector using virtual functions, e.g. element access or insertion is a virtual function call. When using the template function as above, this virtual function call is not necessary, instead the actual member functions of MVKVectorInline or MVKVectorDefault are called. The performance implications are absolutely negligible, so its fine to use the original version instead of my template version.

So feel free to revert it back to your original version, if you prefer that.

Reply:
As mentioned...I tried reverting it back to `MVKVector<uint32_t>&` as part of this...but that causes strange compile errors in the `updateSwizzle()` function about `resize()` not being recognized...and the assignment operator not being valid.

Reply:
Yes sorry, there was a reason for adding the template. The MVKVector class does not offer all member functions. I wanted to avoid all those virtual function calls ( again no real performance implication ). If desired, I will add all those functions to this class to make the usage of MVKVector simpler.

Reply:
Hmmm...yeah...if this is going to be the integral part of MoltenVK that is now is...I think you'll need to add/adjust functions to make it compatible with `std::vector`.

Not a huge priority...since it's functional now...but when you have spare cycles...maybe add bits and pieces as you go (start with `resize()` and assignment to get that `updateSwizzle()` working). 

Otherwise it looks like we'll end up littering the MoltenVK codebase with unnecessary template functions.